src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -372,6 +381,84 @@ impl HummockVersionUpdateExt for HummockVersion {
         split_id_vers
     }
 
+    fn build_sst_delta_infos(&self, version_delta: &HummockVersionDelta) -> Vec<SstDeltaInfo> {
+        let mut infos = vec![];
+
+        for (group_id, group_deltas) in &version_delta.group_deltas {
+            let mut info = SstDeltaInfo::default();
+
+            let mut removed_l0_ssts: BTreeSet<u64> = BTreeSet::new();
+            let mut removed_ssts: BTreeMap<u32, BTreeSet<u64>> = BTreeMap::new();
+
+            // Build only if all deltas are intra level deltas.
+            if !group_deltas
+                .group_deltas
+                .iter()
+                .all(|delta| matches!(delta.get_delta_type().unwrap(), DeltaType::IntraLevel(..)))
+            {
+                continue;
+            }
+
+            for group_delta in &group_deltas.group_deltas {
+                if let DeltaType::IntraLevel(delta) = group_delta.get_delta_type().unwrap() {
+                    if !delta.inserted_table_infos.is_empty() {
+                        info.insert_sst_level = delta.level_idx;
```

Comment:
Will it overwrite the previously set value?

Reply:
Nope, all operations here only requrie immutable reference. Nothing will be changed.

Reply:
I get you. I'll add comments here.